EIPS/eip-minimal-proxy-push0.md

```diff
@@ -0,0 +1,251 @@
+---
+title: Minimal Proxy Contract with `PUSH0`
+description: Optimize the previous Minimal Proxy Contract (eip-3855) with newly introduced `PUSH0` opcode
+author: 0xAA (@AmazingAng)
+discussions-to: https://ethereum-magicians.org/t/proposal-for-a-new-eip-minimal-proxy-contract-with-push0/15662
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-09-04
+requires: 7, 211, 1167, 3855
+---
+
+## Simple Summary
+
+With the newly introduced `PUSH0` opcode ([eip-3855](https://eips.ethereum.org/EIPS/eip-3855)) at Shanghai Upgrade, we minimized the previous Minimal Proxy Contract ([eip-1167](https://eips.ethereum.org/EIPS/eip-1167)) by 200 gas at deployment and 5 gas at runtime, while remain the same functionalities.
+
+## Abstract
+
+Use `PUSH0` opcode minimize gas cost of the previous Minimal Proxy Contract, which simply and cheaply clone contract functionality in an immutable way.
```

Comment:
```suggestion
Use `PUSH0` opcode to minimize gas cost of the previous Minimal Proxy Contract, which simply and cheaply clone contract functionality in an immutable way.
```

Reply:
is `which simply and cheaply clone contract functionality in an immutable way.` this meant for use of push0 or the minimal contract. language is not clear

Reply:
minimal proxy contract